src/CustomerIOInAppMessaging.tsx

```diff
@@ -0,0 +1,75 @@
+import { NativeEventEmitter, NativeModules, Platform } from 'react-native';
+
+const LINKING_ERROR =
+  `The package 'customerio-reactnative' doesn't seem to be linked. Make sure: \n\n` +
+  Platform.select({ ios: "- You have run 'pod install'\n", default: '' }) +
+  '- You rebuilt the app after installing the package\n' +
+  '- You are not using Expo managed workflow\n';
+
+/**
+ * Get CustomerioInAppMessaging native module
+ */
+const InAppMessagingNative = NativeModules.CustomerioInAppMessaging
+  ? NativeModules.CustomerioInAppMessaging
+  : new Proxy(
+    {},
+    {
+      get() {
+        throw new Error(LINKING_ERROR);
```

Comment:
React native package includes in-app module by default. Customers can only disable it but not exclude the module. Also the classes it requires are included in react native package (i.e. `RNCIOInAppMessaging.kt` and `CustomerioInAppMessaging.swift`). So this can only happen if the linking breaks completely. And that is not ideal for customers because our SDK integration will already be broken at all places of interaction.

Reply:
Ok. I can see how an exception being thrown could be useful in that scenario. 

However, what if in the future we changed the RN SDK so the in-app module was not installed by default? We would need to remember to change this line of code, right? 

I prefer not to write code which has consequences if you forget to do something (I tend to forget stuff easily). 

However, now that I look over this code again, the in-app functionality is contained in 1 file right now. If we were to modify the in-app module being installed by default, this code could more easily be discovered by someone.

Reply:
This is not going to break things even then, as the class is just bridge for communication between react native class and wrapper SDKs.

If somehow, we decide not to include in-app by default, that will be a big change in terms of impact. I don't think any change with it should go unnoticed. So I'm keeping this now.